tests/python/unittest/test_sparse_ndarray.py

```diff
@@ -735,6 +735,58 @@ def test_sparse_nd_exception():
     assert_exception(mx.nd.sparse.zeros, ValueError,
                      "invalid_stype", (2,2))
 
+def test_sparse_nd_check_format():
+    """ test check_format for sparse ndarray """
+    def check_csr_format_exception(data_list, indices_list, indptr_list, shape):
```

Comment:
We can use assertRaises to make the code cleaner 
https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/common.py#L27